src/Paramore.Brighter/TransformPipelineBuilder.cs

```diff
@@ -0,0 +1,262 @@
+                throw new ConfigurationException("Error building wrap pipeline for outgoing message, see inner exception for details", e);
+            }
+        }
+
+        public UnwrapPipeline<TRequest> BuildUnwrapPipeline<TRequest>() where TRequest : class, IRequest
+        {
+            try
+            {
+                var messageMapper = FindMessageMapper<TRequest>();
+
+                var transforms = BuildTransformPipeline<TRequest>(FindUnwrapTransforms(messageMapper));
+
+                var pipeline = new UnwrapPipeline<TRequest>(transforms, _messageTransformerFactory, messageMapper);
+
+                s_logger.LogDebug(
+                    "New unwrap pipeline created for: {message} of {pipeline}", typeof(TRequest).Name,
+                    TraceUnwrapPipeline(pipeline)
+                );
+                
+                var wraps = FindWrapTransforms(messageMapper);
+                if (wraps.Any())
+                {
+                    s_logger.LogDebug(
+                        "Wrap attributes on MapToRequest method for mapper of: {message} in {pipeline}, will be ignored", typeof(TRequest).Name,
+                        TraceUnwrapPipeline(pipeline)
+                    );
+                }
+
+                return pipeline;
+            }
+            catch (Exception e)
+            {
+                throw new ConfigurationException("Error building unwrap pipeline for outgoing message, see inner exception for details", e);
+            }
+        }
+
+        private IEnumerable<IAmAMessageTransformAsync> BuildTransformPipeline<TRequest>(IEnumerable<TransformAttribute> transformAttributes)
+            where TRequest : class, IRequest
+        {
+            var transforms = new List<IAmAMessageTransformAsync>();
+
+            //Allowed to be null to avoid breaking v9 interfaces
+            if (_messageTransformerFactory == null)
+            {
+                int i = transformAttributes.Count();
+                if (i > 0)
+                    s_logger.LogWarning(
+                        "No message transformer factory configured, so no transforms will be created but {transformCount} configured", i);
+
+                return transforms;
+            }
+
+            transformAttributes.Each((attribute) =>
+            {
+                var transformType = attribute.GetHandlerType();
+                var transformer = new TransformerFactory<TRequest>(attribute, _messageTransformerFactory).CreateMessageTransformer();
+                if (transformer == null)
+                {
+                    throw new InvalidOperationException(string.Format("Message Transformer Factory could not create a transform of type {0}",
+                        transformType.Name));
+                }
+                else
+                {
+                    transforms.Add(transformer);
+                }
+            });
+
+            return transforms;
+        }
+
+        public static void ClearPipelineCache()
```

Comment:
This doesn't appear used, given that this is somewhat internal, I suppose being static it could be useful

Reply:
It is for tests, though it has not been needed at this point.